crates/chia-consensus/src/gen/build_compressed_block.rs

```diff
@@ -0,0 +1,417 @@
+            num_skipped: 0,
+            ser,
+        })
+    }
+
+    /// add a spend bundle to the generator. The cost must be *only* the CLVM
+    /// execution cost + the cost of the conditions. It must not include the byte cost
+    /// of the bundle. The byte cost is unpredictible as the generator is being
+    /// compressed. The true byte cost is computed by this function.
+    /// returns true if this bundle could be added to the generator, false otherwise
+    pub fn add_spend_bundle(
+        &mut self,
+        bundle: &SpendBundle,
+        cost: u64,
+        constants: &ConsensusConstants,
+    ) -> io::Result<(bool, BuildBlockResult)> {
+        // if we're very close to a full block, we're done. It's very unlikely
+        // any transaction will be smallar than MIN_COST_THRESHOLD
+        if self.byte_cost + self.block_cost + MIN_COST_THRESHOLD >= constants.max_block_cost_clvm {
+            self.num_skipped += 1;
+            return Ok((false, BuildBlockResult::Done));
+        }
+
+        if self.byte_cost + self.block_cost + cost >= constants.max_block_cost_clvm {
+            self.num_skipped += 1;
+            return Ok((false, result(self.num_skipped)));
+        }
+
+        let a = &mut self.allocator;
+
+        let mut spend_list = self.sentinel;
+        for spend in &bundle.coin_spends {
+            // solution
+            let solution = node_from_bytes_backrefs(a, spend.solution.as_ref())?;
+            let item = a.new_pair(solution, NodePtr::NIL)?;
+            // amount
+            let amount = a.new_number(spend.coin.amount.into())?;
+            let item = a.new_pair(amount, item)?;
+            // puzzle reveal
+            let puzzle = node_from_bytes_backrefs(a, spend.puzzle_reveal.as_ref())?;
+            let item = a.new_pair(puzzle, item)?;
+            // parent-id
+            let parent_id = a.new_atom(&spend.coin.parent_coin_info)?;
+            let item = a.new_pair(parent_id, item)?;
+
+            spend_list = a.new_pair(item, spend_list)?;
+        }
+
+        let (done, state) = self.ser.add(a, spend_list)?;
+        assert!(!done);
+
+        self.byte_cost = (self.ser.size() + 2) * constants.cost_per_byte;
+        // closing the lists at the end needs 2 extra bytes
+        if self.byte_cost + self.block_cost + cost > constants.max_block_cost_clvm {
+            // undo the last add() call
+            self.ser.restore(state);
+            self.byte_cost = (self.ser.size() + 2) * constants.cost_per_byte;
+            self.num_skipped += 1;
+            return Ok((false, result(self.num_skipped)));
+        }
+        self.block_cost += cost;
+        self.signature.aggregate(&bundle.aggregated_signature);
+
+        // if we're very close to a full block, we're done. It's very unlikely
+        // any transaction will be smallar than MIN_COST_THRESHOLD
+        let result = if self.byte_cost + self.block_cost + MIN_COST_THRESHOLD
+            >= constants.max_block_cost_clvm
```

Comment:
Same question as above about whether we're intentionally changing behavior to disallow values equal to the maximum allowed.